x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -702,4 +706,13 @@ private static Expression stringToDate(Expression stringExpression) {
             return result;
         }
     }
+
+    private BitSet gatherPreAnalysisMetrics(LogicalPlan plan, BitSet b) {
+        // count only the explicit "limit" the user added, otherwise all queries will have a "limit" and telemetry won't reflect reality
+        if (plan.collectFirstChildren(Limit.class::isInstance).isEmpty() == false) {
+            b.set(LIMIT.ordinal());
+        }
+        plan.forEachDown(p -> FeatureMetric.set(p, b));
```

Comment:
To be clean and not have specific checks for `rename`, `drop`, `keep` (like it did in the first version of the code), this is more like a shotgun approach and many metrics will probably "check" from this step.